File: PO PROJEKT 1/PO PROJEKT 1/PROJEKT PO PLIKI ŹRÓDŁOWE/Sheep.cpp
#include "Sheep.h"

Sheep::Sheep(World* world) {
	increaseStrength(4);
	setInitiative(4);
	sprite = 'o';
	this->world = world;
	spawn();
}

Sheep::Sheep(World* world, Coordinate coord) {
	increaseStrength(4);
	setInitiative(4);
	sprite = 'o';
	this->world = world;
	world->board.insert({ coord, this });
}

Organism* Sheep::createChild(Coordinate coord) {
	return new Sheep(world, coord);
}

File: Symulator Świata/sheep.py
from animal import Animal


class Sheep(Animal):

    def __init__(self, world, coord=None):
        self.name = 'o'
        self._strength = 4
        self._age = 0
        self._initiative = 4
        self.sprite = (255, 255, 255)
        self.world = world
        if coord is not None:
            world.board[coord] = self
        else:
            self.spawn()

    def createChild(self, coord):
        sheep = Sheep(self.world, coord)
